# Análisis Emergencias ambientales asociadas a sustancias químicas

El objetivo es cargar y explorar la base de datos publica que se obtuvo de la Institución: Procuraduría Federal de Protección al Ambiente (PROFEPA)

Trabajaremos con información registrada **del 1° al 3° trimestre de 2025**, lo cual permitirá analizar las emergencias ambientales de ese periodo.

Para ello trabajare con el datasets:  

- **07_Emergencias_ambientales_2025.csv** → información de los estado del país México y las emergencia ambientales suscitadas (derrames, explosion, incendio, fugas, otros)  

Actividades **explorar**, **limpiar** y **analizar** estos datos para construir un **analisis estadístico** de los emergencias ambientales, ocurridas por **entidad federativa** y por **el tipo de emergencia**.  

Este análisis permitirá **identificar entidades con mayores emergencias ambientales** y **diseñar estrategias de prevención**



---
## 🧩 Paso 1: Cargar y explorar

Antes de limpiar o combinar los datos, es necesario **familiarizarte con la estructura del datasets**.  
En esta etapa, valido que el archivo se cargue correctamente, para conocer sus columnas y tipos de datos, y detectar posibles inconsistencias.

### 1.1 Carga de datos y vista rápida

**🎯 Objetivo:**  
Tener **el dataset listo en memoria**, entender su contenido y realizar una revisión preliminar.



In [ ]:
# importar librerías
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
# cargar archivos
emergencia= pd.read_csv('/content/07_Emergencias_ambientales_2025.csv')

In [ ]:
emergencia.head( ) # mostrar las primeras 5 filas de usage

,estado,derrame,explosion,fuga,incendio,otro
0,Aguascalientes,0,1,2,1,0
1,Baja California,2,3,6,1,1
2,Baja California Sur,6,0,0,0,0
3,Campeche,2,0,2,0,0
4,Chiapas,0,3,0,2,0


### 1.2 Exploración de la estructura de los datasets

**🎯 Objetivo:**  
Conocer la **estructura del dataset**, revisar cuántas filas y columnas tienen, identificar los **tipos de datos** de cada columna y detectar posibles **inconsistencias o valores nulos** antes de iniciar el análisis.

**Instrucciones:**  
- Revisa el **número de filas y columnas** del dataset usando `.shape`.  
- Usa `.info()` en cada DataFrame para obtener un **resumen completo** de columnas, tipos de datos y valores no nulos.  

In [ ]:
# revisar el número de filas y columnas del dataset
emergencia.shape


(32, 6)

In [ ]:
emergencia.info() # inspección de emergencia con .info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   estado     32 non-null     object
 1   derrame    32 non-null     int64 
 2   explosion  32 non-null     int64 
 3   fuga       32 non-null     int64 
 4   incendio   32 non-null     int64 
 5   otro       32 non-null     int64 
dtypes: int64(5), object(1)
memory usage: 1.6+ KB


---

## 🧩Paso 2: Identificación de problemas de calidad de datos

### 2.1 Revisión de valores nulos

**🎯 Objetivo:**  
Detectar la presencia y magnitud de valores faltantes para evaluar si afectan el análisis o requieren imputación/eliminación.



In [ ]:
# Validar que no existan estado con falta de ortografia o repetidos
emergencia['estado'].value_counts()

,count
estado,
Aguascalientes,1
Baja California,1
Baja California Sur,1
Campeche,1
Chiapas,1
Chihuahua,1
Ciudad de Mexico,1
Coahuila,1
Colima,1


In [ ]:
# cantidad de nulos

print(emergencia.isna().mean().sort_values(ascending=False)*100) # Proporción de valores nulos)

estado       0.0
derrame      0.0
explosion    0.0
fuga         0.0
incendio     0.0
otro         0.0
dtype: float64


In [ ]:
# Verificar si hay filas duplicadas
emergencia.duplicated().sum()



np.int64(0)

In [ ]:

# Realizar limpieza a columna estado
emergencia['estado']=emergencia['estado'].str.strip().str.title()



---

## 🧩Paso 3: Resumen estadistico por emergencia ambiental


### 3.1 Agrupación por emergencia ambiental

🎯**Objetivo**: Resumir las variables clave de la tabla **emergencia ambiental**, creando métricas que representen su comportamiento real.



In [ ]:
# suma de emergencia del 1° al 3° trimestre de 2025
emergencia[['derrame', 'explosion', 'fuga', 'incendio', 'otro']].sum()

,0
derrame,178
explosion,152
fuga,184
incendio,102
otro,10


In [ ]:
# Emergencias por estado
emergencias_por_estado = (
    emergencia.groupby('estado')[['derrame', 'incendio', 'explosion','fuga','otro']]
    .sum()
)

emergencias_por_estado['total_emergencias'] = (
    emergencias_por_estado['derrame'] +
    emergencias_por_estado['incendio'] +
    emergencias_por_estado['explosion'] +
    emergencias_por_estado['fuga'] +
    emergencias_por_estado['otro']
)

emergencias_por_estado = emergencias_por_estado.sort_values(
    by='total_emergencias',
    ascending=False
)

emergencias_por_estado.head()

,derrame,incendio,explosion,fuga,otro,total_emergencias
estado,,,,,,
Ciudad de Mexico,5,4,15,63,1,88
Mexico,15,16,39,7,1,78
Nuevo Leon,13,15,10,9,0,47
Veracruz,21,9,5,8,2,45
Jalisco,18,9,7,6,1,41


In [ ]:
# Estado con mas emergencias ambientales
estado_max = emergencias_por_estado['total_emergencias'].idxmax()

print(f"El estado con la mayor cantidad de emergencias es: {estado_max}")


El estado con la mayor cantidad de emergencias es: Ciudad de Mexico


In [ ]:
# Estado con menos emergencias ambientales
estado_min = emergencias_por_estado['total_emergencias'].idxmin()

print(f"El estado con la menor cantidad de emergencias es: {estado_min}")


El estado con la menor cantidad de emergencias es: Durango


In [ ]:

# Estado con mas derrames
estado_derrame = emergencias_por_estado['derrame'].idxmax()
print(f"El estado con la mayor cantidad de derrames es: {estado_derrame}")

El estado con la mayor cantidad de derrames es: Veracruz


In [ ]:
# Estado con mas incendios
estado_incendio = emergencias_por_estado['incendio'].idxmax()
print(f"El estado con la mayor cantidad de incendios es: {estado_incendio}")

El estado con la mayor cantidad de incendios es: Mexico


In [ ]:
# Estado con mas explosiones
estado_explosion = emergencias_por_estado['explosion'].idxmax()
print(f"El estado con la mayor cantidad de explosiones es: {estado_explosion}")

El estado con la mayor cantidad de explosiones es: Mexico


In [ ]:
# Estado con mas fugas
estado_fugas = emergencias_por_estado['fuga'].idxmax()
print(f"El estado con la mayor cantidad de fugas es: {estado_fugas}")

El estado con la mayor cantidad de fugas es: Ciudad de Mexico


In [ ]:
#indice de diversidad de emergencias por estado (el estado tiene muchos tipos de problemas o solo uno)
import numpy as np

def diversidad(row):
    p = row / row.sum()
    return -(p * np.log(p + 1e-9)).sum()

emergencias_por_estado['diversidad'] = emergencias_por_estado[['derrame','incendio','explosion','fuga','otro']].apply(diversidad, axis=1)
print(emergencias_por_estado)

                     derrame  incendio  explosion  fuga  otro  \
estado                                                          
Ciudad de Mexico           5         4         15    63     1   
Mexico                    15        16         39     7     1   
Nuevo Leon                13        15         10     9     0   
Veracruz                  21         9          5     8     2   
Jalisco                   18         9          7     6     1   
Puebla                     5         4         10    22     0   
Hidalgo                   20         7          9     3     0   
Guanajuato                13         5          5     2     1   
Tamaulipas                 9         5          4     7     0   
Queretaro                  7         5          3     6     0   
Michoacan                  3         1          5     6     0   
Coahuila                   3         3          1     4     2   
Baja California            2         1          3     6     1   
Oaxaca                   


---
## 🧩Paso 4: Insight Ejecutivo para Stakeholders

🎯 **Objetivo:** Traducir los hallazgos del análisis en conclusiones accionables para el negocio, enfocadas en segmentación, patrones de uso y oportunidades comerciales.



### Análisis ejecutivo

⚠️ **Problemas detectados en los datos**
- En el dataset no se obsevaron valores duplicados, nulos, con falta de ortografia, se encontraron los datos limpios.


🔍 **Analisis de emergencias ambientales**
- Se detecto que en los 3 trimestres del año 2025 en el país México las emergencis ambientales que mas se presentaron fueron 184 fugas, seguido de 178 derrames, 152 explosiones.


📊 **Segmentos por Estado**
- La ciudad de México es el estado con más emergencias ambientales registradas, esto se puede deber a multiples factores, sin embargo, la emergencia ambiental mas predominante fue las fugas.
-Se detecto que el estado con mas derrames es el estado de Veracruz
-El estado con mas incendios fue el estado de México
-El estado con mas explosiones fue el estado de México
-El estado con mas fugas fue la ciudad de México
-En el top 3 de estados con mayor indice de emergencia ambientales esta: Ciudad de México, Estado de México y Nuevo León, estos tres estados son donde existe mayor concentranción de habitantes y por lo tanto mayor numero de empresas.
-Aunque la Ciudad de México obtuvo el primer lugar como estado con mayor número de emergencias ambientales es importante recalcar que el Estado de México es quien mayor diversidad de emergencias tiene, es decir, es quien lidera el ranking en 3 emergencias ambientales (derrame, incendio, explosión), mientras la ciudad de México solo lidera en fugas


💡 **Recomendaciones**
- Las empresas y obras en construcción deben comprometerse mas con el cuidado del medio ambiente, por lo que se debe reforzar las inspecciones en campo por parte de las instituciones gubernamentales, para identificar si se les esta dando cumplimiento a las leyes y normativas ambientales.
- Pos su parte se recomienda que las empresas u obras de  contrucción realicen reportes periodicos a la institucion gubernamental sobre las medidas de prevención y mitigación para evitar que se presenten emergencias ambientales, esto puede ser por medio de una plataforma nacional, en donde se recopile la información, tomando como puntos de interes aquellas empresas que no envien información o no cumplan con las leyes o normas, para inspecciones mas puntuales.

---

## 🧩Paso 5: Visualización de emergencias ambientales


🎯 **Objetivo:**  
Entender visualmente cómo se comportan los estados de acuerdo al **tipo de emergencia ambiental**

Se realiza un dashboard con la informacion obtenida para entender visualmente el comportamiento de los estados con repecto a las emergencias ambientales que se han presentado en los 3 trimestres de 2025


In [ ]:
from google.colab import files

emergencia.to_csv('emergencias.csv', index=False)
files.download('emergencias.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>